.github/workflows/actions/compile-gh-pages/action.yml

```diff
@@ -0,0 +1,32 @@
+name: Compile GitHub Pages
+description: Compiles content for GitHub Pages and uploads it to the artifact cache
+
+inputs:
+  pr-number:
+    type: number
+  repository:
+    default: 'solana-web3.js'
+    type: string
+
+runs:
+  using: composite
+  steps:
+    - name: Compile
+      shell: bash
+      run: pnpm turbo run compile:typedefs compile:ghpages --concurrency=${TURBO_CONCURRENCY:-1}
+      env:
+        REACT_EXAMPLE_APP_BASE_PATH: /${{ inputs.repository }}${{ inputs.pr-number && format('/{0}', inputs.pr-number) || '' }}/example/
+
+    - name: Assemble Deploy Directory
+      shell: bash
+      run: |
+        mkdir -p .ghpages-deploy
+        mv ./examples/react-app/dist/ .ghpages-deploy/example/
+        mv ./docs/* .ghpages-deploy/
+
+    - uses: actions/upload-artifact@v4
+      with:
+        include-hidden-files: true
+        name: ghpages-deploy-artifacts
+        path: .ghpages-deploy
+        retention-days: 1
```

Comment:
Is this going to delete the preview if it's not reviewed within 1 day? I guess we could just run the job again, but is there a downside to making this a bit longer?

Reply:
Because all of these CI workflows are stateless, the way that they assemble the GitHub Pages site is by uploading them to a temporary ‘artifacts cache’ that only needs to live for the duration of the CI run. 1 day is actually too long.